assets/hip-1037/Specification-Format-Style-Guidelines.md

```diff
@@ -0,0 +1,408 @@
+# Protocol Buffer Style Guidelines
+## Style Guidelines for "specification" comments
+### General Style
+- Wrapping is 80 characters for top-level messages.
+- Wrap everything else to 80 characters or 100, depending
+   - Prefer to wrap lines to 80 characters
+   - It is OK to go over 80, but less than 100, if it makes the
+     text clearer and easier to read.
+     - This SHOULD be rare, however.
+   - If you have links or tags, wrapping at 100 characters for clarity
+     is acceptable.
+   - We choose 80 characters here to work better in reviews because Github
+     comment boxes are limited to roughly 80 characters width. The primary
+     "code" interaction with proto files and markdown files is in review.
+     Also, some markdown processors struggle with longer lines.
+- Use HTML tags where appropriate, but stick to markdown as much as possible.
+- Use blank lines (with `*` prefix) for paragraph breaks in message or
+  enum declarations, regardless of where defined.
+   - Use `<p>` tags for _fields_, and _no_ "empty" lines (it breaks the tables).
+   - For messages (including `enum` or `service` messages), we prefer to use
+     blank lines exclusively, and _not_ use a `<p>` tag.
+- We use line breaks (`<br/>`) in several areas of the specification text.
+  All such line breaks should be placed at the _end_ of the prior line, and
+  should not be placed alone on a line, in the middle of a line, or at the
+  beginning of a line.
+- Certain elements (headings, code blocks, lists, etc...) are "block" elements
+  in markdown and HTML.  Such elements _should not_ be followed by either
+  a break (`<br/>`) or paragraph (`<p>`) tag (or equivalent markdown).
+- Use formal US-English spelling and grammar to the maximum extent possible.
```

Comment:
minor: Do we need to require locale?
```suggestion
- Use formal English spelling and grammar to the maximum extent possible.
```

Reply:
Unless we want constant arguments over spelling of certain words, it's highly recommended to choose which version of English is used.

Reply:
Different locale spelling I don't mind as much (e.g. behaviour vs behavior) because contributors will come from all over the world but the others I agree can cause confusion so it's a fair consideration.